src/fluent/watch.ts

```diff
@@ -361,17 +377,43 @@ export class Watcher<T extends GenericClass> {
 
       // Build the URL and request options
       const { opts, url } = await this.#buildURL(true, this.#resourceVersion);
+      let agentOptions;
+      if (opts.agent && opts.agent instanceof https.Agent) {
+        agentOptions = {
+          key: opts.agent.options.key,
+          cert: opts.agent.options.cert,
+          ca: opts.agent.options.ca,
+          rejectUnauthorized: false,
+        };
+      }
 
-      // Create a stream to read the response body
-      this.#stream = byline.createStream();
-
-      // Bind the stream events
-      this.#stream.on("error", this.#errHandler);
-      this.#stream.on("close", this.#streamCleanup);
-      this.#stream.on("finish", this.#streamCleanup);
+      const agent = new Agent({
+        // https://github.com/nodejs/undici/blob/87d7ccf6b51c61a4f4a056f7c2cac78347618486/docs/docs/api/Errors.md?plain=1#L16
+        // https://github.com/nodejs/undici/blob/87d7ccf6b51c61a4f4a056f7c2cac78347618486/docs/docs/api/Client.md?plain=1#L24
+        keepAliveMaxTimeout: 600000,
+        keepAliveTimeout: 600000,
+        bodyTimeout: 0,
+        connect: {
```

Comment:
could you not just set `connect: agentOptions` here?

Reply:
almost 100% sure that the answer is yes. I was debugging bc I had run into an issue and needed to put breakpoints on some lines. Will correct that.